rfd/9999-aws-api-integration-using-oidc.md

```diff
@@ -0,0 +1,554 @@
+                        │  │           └┬───────┼──┘                  │
+                        │  │            │       │                     │
+                        │  │           4│       │                     │
+                        │  │            │       │                     │
+                        │  │  ┌─────────▼┐      │                     │
+                        │  │  │ CA KeySet│      │                     │
+                        │  │  │ +OIDC CA │      │                     │
+                        │  │  └─────────▲┘      │                     │
+                        │  │6           │7      │                     │
+                        │  │     ┌──────┴────┐  │                     │
+                        │  │     │ Discover  │  │           8         │
+                        │  └────►│ Service   ├──┼─────────────────────┘
+                        │        │ [ RDS ]   │  │
+                        │        └───────────┘  │
+                        │                       │
+                        └───────────────────────┘
+```
+
+#### User's Point of View
+When the user adds a new resource, using the Discover Wizard, they'll be asked where the resource lives.
+Choosing an RDS DB, the user is presented with the following flow:
+
+1. User is prompted to setup an OIDC Identity Provider pointing to its Teleport instance's url.
+2. User opens AWS, selects IAM and then, under `Access Management`, selects `Identity providers`.
+3. Configures the Teleport instance as an OIDC IdP:
+   1. Clicks `Add provider` of type `OpenID Connect`.
+   2. Fills in the Provider URL as provided by the Discover Wizard.
+   3. Fills in the Audience as provided by the Discover Wizard `sts.amazonaws.com`.
+4. The user is then asked to enter the Thumbprint in Teleport, to ensure the user entered the correct `Provider URL`.
+5. Configures the role associated with that IdP
+   1. The user is asked to create or assign an existing role.
+   2. The role's `Policies` must be filled with the required Policies (described down below).
+   3. The role's `Trust relationships` must target the Identity Provider they just created (described down below).
+6. The user will be greeted with a message in Teleport saying that the configuration was successful.
+
+To ease the necessary manual steps, the Discover wizard will provide a small video demoing what the user must do for the previous steps.
+
+At this point, the integration is configured and the user is able to discover RDS DBs without leaving Teleport.
+
+#### System's Point of View
+The first interaction between AWS and Teleport happens when the user clicks on `Get Thumbprint` (step 4 of the User's Point of View flow).
+This will trigger a request started by AWS to Teleport hitting the OpenID Configuration endpoint and subsequently the JSON Web Key Set endpoint.
+
+After this step, when the user tries to list RDS DBs, Teleport generates a token and issues the API call.
```

Comment:
Can you have multiple AWS integrations e.g. for different accounts? How do we pick which one to use?

Reply:
Yes, multiple AWS Integrations must be supported.
I'll add that.

I'll also add that the user must be able to select which AWS Integration to use.